dep0/app/dep0/main.cpp

```diff
@@ -1,187 +1,188 @@
+            cl::cat(mainCat),
+            cl::desc(
+                "Compile only and emit unverified LLVM IR code; but do not assemble or link.\n"
+                "This is only useful when debugging the llvmgen module"));
+    auto const typecheck_only =
+        cl::opt<bool>(
+            "t",
+            cl::init(false),
+            cl::cat(mainCat),
+            cl::desc("Typecheck only; do not compile, assemble or link"));
+
+    // extra options - sorted by the actual command line argument
+    cl::OptionCategory extraCat("Secondary Options", "These are extra options that can be useful occasionally");
     auto const mtriple =
         cl::opt<std::string>(
             "mtriple",
-            cl::desc("Override target triple for module"));
+            cl::cat(extraCat),
+            cl::desc("Override the target triple used during compilation and assembly stages"));
     auto const no_prelude =
         cl::opt<bool>(
             "no-prelude",
-            cl::desc("Do not pre-import the prelude module"),
-            cl::init(false));
+            cl::init(false),
+            cl::cat(extraCat),
+            cl::desc("Do not pre-import the prelude module"));
     auto const print_ast =
         cl::opt<bool>(
             "print-ast",
-            cl::desc("Print the AST before the codegen stage"),
-            cl::init(false));
+            cl::init(false),
+            cl::cat(extraCat),
+            cl::desc(
+                "When using -t, print the resulting AST.\n"
+                "The AST is printed after the transformation stage, unless --skip-transformations is also set"));
     auto const skip_transformations =
         cl::opt<bool>(
             "skip-transformations",
-            cl::desc("Skip the transformations pipeline stage"),
-            cl::init(false));
-    auto typecheck_only =
-        cl::opt<bool>(
-            "typecheck-only",
-            cl::desc("Stop after typechecking; don't emit any asm/obj code"),
-            cl::init(false));
-    auto const alias_t = cl::alias("t", cl::desc("alias for --typecheck-only"), cl::aliasopt(typecheck_only));
-    auto unverified =
-        cl::opt<bool>(
-            "unverified",
-            cl::desc("Emit unverified LLVM IR; only useful when debugging the llvmgen module"),
-            cl::init(false));
-    auto const alias_u = cl::alias("u", cl::desc("alias for --unverified"), cl::aliasopt(unverified));
+            cl::init(false),
+            cl::cat(extraCat),
+            cl::desc("Skip the transformations pipeline stage, for example beta-delta normalization"));
 
-    llvm::InitLLVM LLVM(argc, argv);
-    llvm::InitializeAllTargets();
-    llvm::InitializeAllTargetMCs();
-    llvm::InitializeAllAsmPrinters();
-    llvm::InitializeAllAsmParsers();
-    cl::ParseCommandLineOptions(argc, argv, "dep0 - DepC Bootstrapping Compiler");
+    auto const input_files = cl::list<std::string>(cl::Positional, cl::desc("<input-files>"));
```

Comment:
What happens with this instead?

```suggestion
    auto const input_files = cl::list<std::filesystem::path>(cl::Positional, cl::desc("<input-files>"));
```

Reply:
I wish... LLVM's command line API is quite disappointing :(